platform/platform-migratable/src/test/java/org/codice/ddf/platform/migratable/impl/PlatformMigratableTest.java

```diff
@@ -0,0 +1,273 @@
+                mockMigratableUtil);
+
+        // Perform Test
+        MigrationMetadata migrationMetadata = platformMigratable.export(exportDirectory);
+
+        // Verify
+        assertThat(migrationMetadata.getMigrationWarnings().size(), is(1));
+        assertThat(migrationMetadata.getMigrationWarnings().contains(expectedMigrationWarning1),
+                is(true));
+    }
+    
+    @Test(expected = MigrationException.class)
+    public void testExportExceptionThrownWhenCopyingFile() throws Exception {
+        // Setup
+        MigratableUtil mockMigratableUtil = mock(MigratableUtil.class);
+        doThrow(MigrationException.class).when(mockMigratableUtil).copyFile(any(Path.class),
+                eq(exportDirectory), Matchers.<Collection<MigrationWarning>> any());
+        PlatformMigratable platformMigratable = new PlatformMigratable(DESCRIPTION, IS_OPTIONAL,
+                mockMigratableUtil);
+
+        // Perform test
+        platformMigratable.export(exportDirectory);
+    }
+
+    @Test(expected = MigrationException.class)
+    public void testExportExceptionThrownWhenCopyingDirectory() throws Exception {
+        // Setup
+        MigratableUtil mockMigratableUtil = mock(MigratableUtil.class);
+        doThrow(MigrationException.class).when(mockMigratableUtil).copyDirectory(any(Path.class),
+                eq(exportDirectory), Matchers.<Collection<MigrationWarning>> any());
+        PlatformMigratable platformMigratable = new PlatformMigratable(DESCRIPTION, IS_OPTIONAL,
+                mockMigratableUtil);
+
+        // Perform test
+        platformMigratable.export(exportDirectory);
+    }
+    
+    @Test(expected = MigrationException.class)
+    public void testExportExceptionThrownWhenCopyingFileFromSystemPropertyValue() throws Exception {
+        // Setup
+        MigratableUtil mockMigratableUtil = mock(MigratableUtil.class);
+        doThrow(MigrationException.class).when(mockMigratableUtil).copyFileFromSystemPropertyValue(
+                any(String.class), eq(exportDirectory),
+                Matchers.<Collection<MigrationWarning>> any());
+        PlatformMigratable platformMigratable = new PlatformMigratable(DESCRIPTION, IS_OPTIONAL,
+                mockMigratableUtil);
+
+        // Perform test
+        platformMigratable.export(exportDirectory);
+    }
+
+    private void assertWsSecurityExport(MigratableUtil mockMigratableUtil) {
+        ArgumentCaptor<Path> sourceDirCaptor = ArgumentCaptor.forClass(Path.class);
+        verify(mockMigratableUtil).copyDirectory(sourceDirCaptor.capture(), eq(exportDirectory),
+                Matchers.<Collection<MigrationWarning>> any());
+        assertThat(sourceDirCaptor.getValue(), is(WS_SECURITY_DIR_REL_PATH));
+    }
+
+    private void assertSystemPropertiesFilesExport(MigratableUtil mockMigratableUtil) {
+        ArgumentCaptor<Path> sourceFileCaptor = ArgumentCaptor.forClass(Path.class);
```

Comment:
Instead of using `ArgumentCaptor`, you should be able to do something like this:

```
verify(mockMigratableUtil).copyFile(SYSTEM_PROPERTIES_REL_PATH, exportDirectory, anyCollectionOf(MigrationWarning.class));
verify(mockMigratableUtil).copyFile(USERS_PROPERTIES_REL_PATH, exportDirectory, anyCollectionOf(MigrationWarning.class));
```

This should also work for the other `assert...()` methods.